### ISSUE TITLE
Unchecking "Automatically check for updates" does not disable check for updates

### ISSUE THREAD

**Issue by GMTA** [docker/for-mac#3908]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

When I uncheck `Automatically check for updates` in the `General` tab of the `Preferences` dialog, I expect no future check for updates to take place.

### Actual behavior

Periodically a check for updates takes place and informs me of the availability of a new version.

### Information
  - macOS Version: Mojave 10.14.6
  - Docker Desktop 2.1.0.2 (stable channel)

### Steps to reproduce the behavior

Difficult to reproduce consistently since the check seems to take place periodically.

**Comment by GMTA** [comment]
### Use case
We fixate software versions to prevent compatibility issues on automatic updates. In the past, Docker Desktop has shown multiple issues in starting up, running containers or accessing the host file system. All of these could have been prevented if we were able to disable the "Automatically check for updates" functionality.

The issue is still present for MacOS Catalina 10.15.1 and Docker Desktop 2.1.0.5.

**Comment by weaver299** [comment]
This behavior has been driving me crazy at least since the time when this ticket was created.  I think I remember it doing this as far back as 2018...
 
 
![Screenshot](https://i.imgur.com/F2F7ZX5.png)
 
 
me: 
> 'automatically check for updates'... I didn't tell it to check.. it seems like it checked 'automatically'.. it had to 'check', or how would it know there's an update... a new version is an 'update' right?  what does this checkbox even do??
 
![Math lady](https://i.kym-cdn.com/entries/icons/mobile/000/021/464/14608107_1180665285312703_1558693314_n.jpg)

**Comment by kmonty** [comment]
Confirming this issue exists for me as well.

**Comment by gingerCodeNinja** [comment]
<img width="1647" alt="Screenshot 2020-09-16 at 11 20 13" src="https://user-images.githubusercontent.com/2407258/93325823-cca84880-f80f-11ea-8d99-fb9493f4065c.png">

Same thing, super annoying. The reason I unchecked this box was to stop this annoying popup in the first place. Also, it'd be good to be in charge of the network traffic apps are using, even if only tiny.

**Comment by geoff-smith** [comment]
This is annoying for me too.

**Comment by dijit** [comment]
This is annoying for me also, this is the top result on google for this problem so if there is a workaround please let me know.

Otherwise I will use a remote docker machine I think.

**Comment by codeseedr** [comment]
I'm also affected (have the checkbox unchecked), in my case I'm getting this annoying popup every now and then.

<img width="710" alt="popup" src="https://user-images.githubusercontent.com/5316283/98854686-2b2e2380-245b-11eb-92db-374498dbd052.png">

**Comment by nnqq** [comment]
+1, how to disable this annoying popup?

**Comment by codeseedr** [comment]
I downgraded to 2.3.5.0 and the popup is gone.

**Comment by nnqq** [comment]
@codeseedr thank you, it help me too!

**Comment by PRKLOCALBROKERS** [comment]
The best way to fix this is to remove the checkbox itself, this is what I see in latest versions (3.1.0). So the updates dialog is now ineludible. 
<img width="1240" alt="Screenshot 2021-01-20 at 1 40 59 AM" src="https://user-images.githubusercontent.com/58592315/105087826-1906dc00-5ac1-11eb-9f48-b19c6360bec9.png">

**Comment by hsarma** [comment]
The best way to fix this is to remove the checkbox itself, this is what I see in latest versions (3.1.0). So the updates dialog is now ineludible.
<img width="1240" alt="Screenshot 2021-01-20 at 1 40 59 AM" src="https://user-images.githubusercontent.com/13164674/105088079-6aaf6680-5ac1-11eb-8134-ba3eed4f134e.png">

**Comment by zeromh** [comment]
I see the same thing. So Docker ignored this issue for over a year, and then removed the option that people wanted. Nice.

**Comment by weaver299** [comment]
> I see the same thing. So Docker ignored this issue for over a year, and then removed the option that people wanted. Nice.

Yeah.. of course implementing the feature properly would have been preferable, but removing the non-functional option is better than leaving it there and broken.  This ticket hasn't been "Closed - Won't Fix" yet so maybe there is still hope for future.

For what it's worth to any devs reading this, I don't use Docker directly, I use a "Docker+" based development environment (Lando) which has been broken in the past by Docker updates, so official recommendation is to not apply Docker upgrades separately from the development environment.  People sometimes come onto slack with broken dev environments because they installed an upstream update with breaking changes...

**Comment by RobinHoutevelts** [comment]
I'm having these super annoying popups. I'm not interested in upgrading!

![snooze-update](https://user-images.githubusercontent.com/9056689/125778670-6361b96f-fd51-47b9-b155-32d4c664b5a9.png)

**Comment by waleed-dawood** [comment]
This is annoying for me too. I have the "Automatically check for updates" unchecked but it still keeps on notifying me about updates. I have attached a screenshot.

This feels like something that should be relatively easy to fix. But after 5 years the issue is still there. 

<img width="494" height="247" alt="Image" src="https://github.com/user-attachments/assets/a2d2b66a-24ef-4e13-a26b-bfd0edc30de8" />

**Comment by mat007** [comment]
@waleed-dawood I suppose you didn’t trigger a check for update manually (by clicking the button in the UI or selecting that entry in the tray menu)?
Can you [create and upload diagnostics](https://docs.docker.com/desktop/troubleshoot-and-support/troubleshoot/#diagnose) and give us the ID?

**Comment by vitaliy-mykytenko** [comment]
Set /etc/hosts "127.0.0.1 desktop.docker.com" to prevent docker check update
